contracts/Exchange.sol

```diff
@@ -235,18 +235,18 @@ contract Exchange is Restricted {
     }
 
     function _removeBuyOrder(Order storage order) private {
-        _removeOrder(activeBuyOrders, order.index);
+        if (order.index < activeBuyOrders.length - 1) {
+            activeBuyOrders[order.index] = activeBuyOrders[activeBuyOrders.length - 1];
+            buyTokenOrders[activeBuyOrders[order.index]].index = order.index;
```

Comment:
might be better to memorize activeBuyOrders[order.index]

Reply:
similarly, `uint lastIndex = activeBuyOrders.length - 1'

Reply:
done